packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -362,4 +343,300 @@ export class UiControlClient extends ApiCommands {
       await this.waitUntil(AskUICommand, maxTry - 1);
     }
   }
+
+  /**
+   * Click a button with a specific label.
+   * Optional relation identifies the button in relation to another element.
+   *
+   * **Examples:**
+   * ```typescript
+   * await aui.clickButton({})
+   * await aui.clickButton({label: 'Checkout here'})
+   * await aui.clickButton({relation: {type: 'leftOf', text: 'Choose a ticket'}})
+   * await aui.clickButton({label: 'Click', {relation: {type: 'leftOf', text: 'Choose a ticket'}})
+   * ```
+   *
+   * @param {Object} params - Object containing properties.
+   * @property {string} [params.label] - The text label of the button. Defaults to an empty string.
+   * @property {Object} [params.relation] - Object describing the relationship between
+   *                                        the clicked button and another element.
+   * @property {Relations} params.relation.type - The type of relation.
+   * @property {string} params.relation.text - The text element the relation is based on.
+   */
+  async clickButton(
+    params: {
+      label?: string,
+      relation?: {
+        type: Relations,
+        text: string
+      }
+    },
+  ) {
+    let command = this.click().button();
+
+    if (params.label) {
+      command = command.withText(params.label);
+    }
+
+    if (params.relation) {
+      switch (params.relation.type) {
+        case 'nearestTo':
+          command = command.nearestTo().text(params.relation.text);
+          break;
+        case 'leftOf':
+          command = command.leftOf().text(params.relation.text);
+          break;
+        case 'above':
+          command = command.above().text(params.relation.text);
+          break;
+        case 'rightOf':
+          command = command.rightOf().text(params.relation.text);
+          break;
+        case 'below':
+          command = command.below().text(params.relation.text);
+          break;
+        default:
+          command = command.nearestTo().text(params.relation.text);
+          break;
```

Comment:
This can only happen if someone is going around the Typescript compiler some way from my perspective using a relation that we don't support. What about throwing an error in this case instead to let the user know so that the user is able to fix this?

Reply:
Good idea. Implemented it this way now :)